File: front-end/src/app/service-models/app/v1/client/client-company-detail-s-m.ts
import { CoinManagementServiceModelBase } from '../../base/coin-management-service-model-base';

export class ClientCompanyDetailSM extends CoinManagementServiceModelBase<number> {
    companyCode!: string;
    name!: string;
    description!: string;
    companyContactEmail!: string;
    companyMobileNumber!: string;
    companyWebsite!: string;
    companyLogoPath!: string;
    isEnabled!: boolean;
    isTrialUsed!: boolean;
    trailLastDate?: Date;
    companyDateOfEstablishment!: Date;
}

File: backend/payroll-backend/Components/Siffrum.Web.Payroll.ServiceModels/v1/Client/ClientCompanyDetailSM.cs
namespace Siffrum.Web.Payroll.ServiceModels.v1.Client
{
    public class ClientCompanyDetailSM : Base.SiffrumPayrollServiceModelBase<int>
    {
        public ClientCompanyDetailSM()
        {
        }

        public string CompanyCode { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string CompanyContactEmail { get; set; }
        //public string ContactEmail { get; set; }
        public string CompanyMobileNumber { get; set; }
        public string CompanyWebsite { get; set; }
        public string CompanyLogoPath { get; set; }
        public bool IsEnabled { get; set; }
        public bool IsTrialUsed { get; set; }
        public DateTime? TrailLastDate { get; set; }

        //[ConvertFilePathToUri(SourcePropertyName = nameof(CompanyLogoPath))]
        public DateTime CompanyDateOfEstablishment { get; set; }
        //public int ClientCompanyAddressId { get; set; }
    }
}
